steering_controllers_library/src/steering_controllers_library.cpp

```diff
@@ -0,0 +1,576 @@
+    {
+      state_interfaces_config.names.push_back(
+        params_.front_wheels_names[i] + "/" + traction_wheels_feedback);
+    }
+
+    for (size_t i = 0; i < params_.rear_wheels_names.size(); i++)
+    {
+      state_interfaces_config.names.push_back(
+        params_.rear_wheels_names[i] + "/" + hardware_interface::HW_IF_POSITION);
+    }
+  }
+
+  return state_interfaces_config;
+}
+
+std::vector<hardware_interface::CommandInterface>
+SteeringControllersLibrary::on_export_reference_interfaces()
+{
+  reference_interfaces_.resize(nr_ref_itfs_, std::numeric_limits<double>::quiet_NaN());
+
+  std::vector<hardware_interface::CommandInterface> reference_interfaces;
+  reference_interfaces.reserve(nr_ref_itfs_);
+
+  reference_interfaces.push_back(hardware_interface::CommandInterface(
+    get_node()->get_name(), std::string("linear/") + hardware_interface::HW_IF_VELOCITY,
+    &reference_interfaces_[0]));
+
+  reference_interfaces.push_back(hardware_interface::CommandInterface(
+    get_node()->get_name(), std::string("angular/") + hardware_interface::HW_IF_POSITION,
+    &reference_interfaces_[1]));
+
+  return reference_interfaces;
+}
+
+bool SteeringControllersLibrary::on_set_chained_mode(bool chained_mode)
+{
+  // Always accept switch to/from chained mode
+  return true || chained_mode;
+}
+
+controller_interface::CallbackReturn SteeringControllersLibrary::on_activate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  // Set default value in command
+  reset_controller_reference_msg(*(input_ref_.readFromRT()), get_node());
+
+  return controller_interface::CallbackReturn::SUCCESS;
+}
+
+controller_interface::CallbackReturn SteeringControllersLibrary::on_deactivate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  // TODO(anyone): depending on number of interfaces, use definitions, e.g., `CMD_MY_ITFS`,
+  // instead of a loop
+  for (size_t i = 0; i < nr_cmd_itfs_; ++i)
+  {
+    command_interfaces_[i].set_value(std::numeric_limits<double>::quiet_NaN());
+  }
+  return controller_interface::CallbackReturn::SUCCESS;
+}
+
+controller_interface::return_type SteeringControllersLibrary::update_reference_from_subscribers()
+{
+  auto current_ref = *(input_ref_.readFromRT());
+  const auto age_of_last_command = get_node()->now() - (current_ref)->header.stamp;
+
+  // TODO(anyone): depending on number of interfaces, use definitions, e.g., `CMD_MY_ITFS`,
+  // instead of a loop
+  // send message only if there is no timeout
+  if (age_of_last_command <= ref_timeout_ || ref_timeout_ == rclcpp::Duration::from_seconds(0))
+  {
+    if (!std::isnan(current_ref->twist.linear.x) && !std::isnan(current_ref->twist.angular.z))
+    {
+      reference_interfaces_[0] = current_ref->twist.linear.x;
+      reference_interfaces_[1] = current_ref->twist.angular.z;
+    }
+  }
+  else
+  {
+    reference_interfaces_[0] = std::numeric_limits<double>::quiet_NaN();
+    reference_interfaces_[1] = std::numeric_limits<double>::quiet_NaN();
```

Comment:
```suggestion
    reference_interfaces_[0] = std::numeric_limits<double>::quiet_NaN();
    reference_interfaces_[1] = std::numeric_limits<double>::quiet_NaN();
```


if (
      !std::isnan(current_ref->twist.linear.x) && !std::isnan(current_ref->twist.angular.z))
    {
      reference_interfaces_[0] = 0.0;
      reference_interfaces_[1] = 0.0;

      current_ref->twist.linear.x = std::numeric_limits<double>::quiet_NaN();
      current_ref->twist.angular.z = std::numeric_limits<double>::quiet_NaN();
    }